pkg/policy/rule.go

```diff
@@ -162,6 +195,12 @@ func mergePortProto(ctx *SearchContext, existingFilter, filterToMerge *L4Filter,
 			// Merge isRedirect flag
 			l7Rules.isRedirect = l7Rules.isRedirect || newL7Rules.isRedirect
 
+			// Merge listener reference
+			if err := l7Rules.mergeListenerReference(newL7Rules); err != nil {
```

Comment:
I think that this is resolving the case where you have two rules selecting the same traffic with different priorities + L7 configuration, eg:

- Traffic to app=foo on port 80/TCP should use listener A with prio 1
- Traffic to app=foo on port 80/TCP should use listener B with prio 2

I think there's another case, which admittedly is probably covered "accurately" by the current API since it's deliberately opaque but consider:

- Traffic to app=foo on port 80/TCP should use listener A with prio 1
- Traffic to class=empire on port 80/TCP should use listener B with prio 1

Given that these are acting on different selectors, I expect them to be encoded in different `*PerSelectorPolicy` objects. However, it's possible for each of these rules to select the same identity if that identity includes both `app=foo`,`class=empire`. In that case, the priority conflict would make the rules impossible to evaluate.

I don't currently see code handling that case. (sidenote, how do we test this?)

Reply:
Thanks for reminding about this. Currently equal priority will revert back to the prior behavior where the result depends on the (random) order of generating the mapstate entries. I'll need to add the use of the proxy port number as a tie-breaker, so it will be deterministic at least.